### DISCUSSION TITLE
Feature Request: Add split_part function as a macro (code below!)

### DISCUSSION THREAD

**Discussion by Alex-Monahan** [duckdb/duckdb#6173] [Ideas]
Hi folks!

Split_part is a function that Postgres has that we can implement with a small wrapper macro. In [this excellent article](https://www.vantage.sh/blog/querying-aws-cost-data-duckdb), split_part was one of their small complaints (in an otherwise positive article!). Here is a macro that I have tested somewhat. It matches Postgres's behavior in that if a `position` is beyond the indexes of the list, it returns an empty string instead of null (https://stackoverflow.com/questions/45766644/substituting-value-in-empty-field-after-using-split-part). 

```sql
CREATE OR REPLACE MACRO split_part(string, delimiter, position) AS
    --The coalesce wrapper is to maintain Postgres behavior compatibility
    coalesce(string_split(string, delimiter)[position],'');
```

```sql
select split_part('a,b,c',',',1);
```
| split_part('a,b,c', ',', 1) |
|-----------------------------|
| a                           |

```sql
select split_part('a,b,c',',',2);
```
| split_part('a,b,c', ',', 2) |
|-----------------------------|
| b                           |

```sql
select split_part('a,b,c',',',-2);
```
| split_part('a,b,c', ',', -2) |
|------------------------------|
| b                            |
```sql
select split_part('a,b,c',',',100);
```
| split_part('a,b,c', ',', 100) |
|-------------------------------|
|                               |

Thanks!
-Alex

**Comment by Mytherin** [comment]
Thanks! That sounds like a great idea indeed. Could you perhaps submit a PR that adds this macro to the [list of built-in macros](https://github.com/duckdb/duckdb/blob/master/src/catalog/default/default_functions.cpp#L13)?

**Reply by Alex-Monahan** [reply]
OOOH, first PR! Let's do this!

**Reply by Alex-Monahan** [reply]
Anywhere that I should add a few tests?

**Reply by Mytherin** [reply]
Yes, you can add your tests to the `test/sql/function/string` folder (create a new file called `test_split_part.test` or so).
